app/src/main/java/com/example/nvdovin/weatherapp/domain/utils/sharedpreferences/SharedPrefs.java

```diff
@@ -13,13 +11,12 @@
     private SharedPreferences sharedPref;
     private SharedPreferences.Editor editor;
 
-    public SharedPrefs(Context context) {
-        sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
-        editor = sharedPref.edit();
+    public SharedPrefs(SharedPreferences preferences) {
+        this.sharedPref = preferences;
+        this.editor = preferences.edit();
```

Comment:
there is no reason to store editor, you can call it in each method. please avoid storing link to unused class.